pkg/util/resolver.go

```diff
@@ -30,6 +31,16 @@ func NewResolver(username, password string, insecure, plainHTTP bool, configs ..
 		opts.Credentials = func(hostName string) (string, string, error) {
 			return username, password, nil
 		}
+		opts.Hosts = config.ConfigureHosts(context.Background(), config.HostOptions{
+			Credentials: func(host string) (string, string, error) {
+				// If host doesn't match...
+				// Only one host
+				return username, password, nil
+			},
+			DefaultTLS: &tls.Config{
+				InsecureSkipVerify: true,
```

Comment:
`insecure` is a bool to this func; I would prefer not setting this to true by default, but use `insecure` here

Reply:
I specify `insecure` then it would skip `https`, my private registry still deploy with TLS.

Reply:
Done.